app/views/edit/build-config.html

```diff
@@ -89,7 +122,7 @@
                               }" ng-model="updatedBuildConfig.spec.source.dockerfile" class="ace-bordered ace-inline dockerfile-mode"></div>
                             </div>
 
-                            <div class="form-group" ng-if="updatedBuildConfig.spec.strategy.dockerStrategy.dockerfilePath">
+                            <div class="form-group" ng-if="updatedBuildConfig.spec.strategy.dockerStrategy.dockerfilePath && advancedOptions">
```

Comment:
Have you confirmed we're not removing an existing dockerfile path if the user doesn't open the advanced options?